nengo/utils/connection.py

```diff
@@ -107,3 +107,21 @@ def eval_point_decoding(conn, sim, eval_points=None):
     decoded = np.dot(activities, weights.T)
     targets = get_targets(conn, eval_points)
     return eval_points, targets, decoded
+
+
+def function_name(func):
+    """Returns the name of a connection function (robust to the different
+    types of objects that can be passed as a function argument).
+
+    Parameters
+    ----------
+    func : callable or array-like
+        object used as function argument
+
+    Returns
+    -------
+    str
+        name of function object
+    """
+
+    return getattr(func, "__name__", func.__class__.__name__)
```

Comment:
Maybe it was for MRO. For instances of old classes, `x.__class__.__mro__` does not exist, whereas `type(x).__mro__` does. However (and maybe I didn't realize this at the time I made the changes), `type(x).__mro__` just gives `(instance, object)` no matter what the actual inheritance is, so it's not useful. Does anybody know how to get the MRO of an old-style class?

Anyway, `x.__class__.__name__` seems generally better to me than `type(x).__name__`, so I'm fine to keep it here. We should also maybe consider reverting 76f0a34